source/simulator/nullspace.cc

```diff
@@ -221,13 +221,6 @@ namespace aspect
 
           }
 
-#ifdef DEBUG
-      Assert(n_left_to_find == 0, ExcMessage("Error, couldn't find a velocity DoF to constrain."));
-
-      for (unsigned int d=0; d<dim; ++d)
-        Assert(vel_idx[d] != numbers::invalid_dof_index,
-               ExcMessage("Error, couldn't find a velocity DoF to constrain."));
-#endif
```

Comment:
...generate CORRECT results?

Thanks for the explanation. If I see this right, the you are using the fact that those processors who compute `vel_idx[d]==invalid_dof_index` will pass `invalid_dof_index` into `Utilities::MPI::min`. This relies on the fact that `invalid_dof_index` is an invalid number that happens to be represented by something large. This is fragile. Could we change this to
```
dealii::Utilities::MPI::min((vel_idx[d] != invalid_dof_index) ?
                                        vel_idx[d] :
                                        dof_handler.n_dofs(),   // a number larger than all DoF indices
                mpi_communicator);
```
or similar? I think that would make it clearer. You'd need to adjust the assertion in line 241 then.